server/src/main/kotlin/com/kamelia/hedera/rest/user/Entity.kt

```diff
@@ -111,7 +111,7 @@ object Users : AuditableUUIDTable("users") {
 
     suspend fun create(user: UserDTO, creator: User? = null): User = Connection.transaction {
         User.new {
-            username = user.username
+            username = user.username.lowercase()
```

Comment:
not sure if this is needed since we precheck anyways, and it only does a partial transformation (as in, it doesn't filter out invalid non letter characters), might as well not have it

Reply:
You are right, this is a forgotten vestige of my first though.

Reply:
https://github.com/Black-Kamelia/Hedera/issues/83#issuecomment-1584522678

i made the concession to restrict the username mainly for performance reasons. so for this time, I think we should actually go for my suggestion to avoid a request and let the db throw.

if you want, I'll look into how to catch the unicity constraint on the name.